storage/store.go

```diff
@@ -989,11 +989,14 @@ func (s *Store) MergeRange(subsumingRng *Replica, updatedEndKey proto.Key, subsu
 
 	// Remove and destroy the subsumed range. Note that we are on the
 	// processRaft goroutine so we can call removeReplicaImpl directly.
-	if err = s.removeReplicaImpl(subsumedRng); err != nil {
+	if err := s.removeReplicaImpl(subsumedRng); err != nil {
 		return util.Errorf("cannot remove range %s", err)
 	}
 
-	// TODO(bram): The removed range needs to have all of its metadata removed.
+	localRangeKeyPrefix := keys.MakeLocalRangeKeyPrefix(subsumedDesc.RangeID)
```

Comment:
Actually, the range descriptor is deleted in the transaction in `AdminMerge` in `replica_command.go`. That's where this should go as well; `Store.MergeRange` shouldn't touch the engine.